src/lib/CollectStore.js

```diff
@@ -434,7 +434,11 @@ export default class CollectStore {
     konnector = this.connectors.find(c => c.slug === konnector.slug)
     return Promise.all(konnector.accounts.map(account => accounts._delete(cozy.client, account)
         .then(() => konnector.accounts.splice(konnector.accounts.indexOf(account), 1))
-        .then(() => konnectors.unlinkFolder(cozy.client, konnector, account.folderId))
+        .then(() =>
+          account.folderId
+            ? konnectors.unlinkFolder(cozy.client, konnector, account.folderId)
+            : Promise.resolve(null)
```

Comment:
I think that we don't need a `Promise.resolve(null)` here, something like 
```javascript
.then(() =>
  if (!account.folderId) return
  return konnector.unlinkFolder(cozy.client, konnector, account.folderId)
)
```
is enough.

Reply:
You're right, thanks 👍